app/views/govuk_publishing_components/components/_radio.html.erb

```diff
@@ -9,6 +9,7 @@
   small ||= false
   inline ||= false
   is_page_heading ||= false
+  heading_size = 'xl' if is_page_heading && !heading_size.present?
```

Comment:
The interplay of these different options (`is_page_heading`, `heading_size`) I'm finding quite confusing. Is there a way of simplifying this? 

Could we have one option for heading (which would take 1, 2, 3 etc. but default to nil, resulting  in no heading) and one option for size (s, m, etc.) and then they wouldn't have to interact?

Reply:
Yeah, that's fine, still feels simpler than the current approach.

Reply:
Correct me if I've misunderstood you but the current solution is `heading_size` and `is_page_heading`. All the above line is doing is checking if a heading size has been set when `is_page_heading` is passed in and if it's not then it's set to `xl` by default.a

Reply:
I've updated the code a little bit - let me know if this is still confusing

Reply:
Sorry, maybe I need more sleep but I'm just finding line 12 hard to read. The heading size is set to xl, unless `is_page_heading` is true and no `heading_size` is specified. Otherwise, on the next line, set heading size to `m` unless a correct heading size has been passed.

If `is_page_heading` is true but no `heading_size` is passed, heading size is set to `m`, is that right? If `is_page_heading` is false but a `heading_size` is passed, it gets set to whatever was passed (if valid).

Could it be rewritten as

```
heading_size = 'm' unless ['s', 'm', 'l', 'xl'].include?(heading_size)
heading_size = 'xl' if is_page_heading
```

Reply:
hey that would not work because we want to change the header size.

We only want to default to `xl` if there isn't a `heading_size` passed in

Reply:
Perhaps it's the 'double assignment' (heading_size getting set to 'm' and then immediately changed to 'xl') that's confusing?

Would something like this work? (not tested!)

```ruby
if ['s', 'm', 'l', 'xl'].include?(heading_size)
  size = heading_size
elsif is_page_heading
  size = 'xl'
else
  size = 'm'
end
```

Or, if you want to avoid the extra variable:

```ruby
unless ['s', 'm', 'l', 'xl'].include?(heading_size)
  if is_page_heading
    heading_size = 'xl'
  else
    heading_size = 'm'
  end
end
```

Reply:
I like the first option. It's more lines of code, but much clearer in its intent.

Reply:
Thanks @36degrees ! I've updated the code.